src/test/arg/basics.spec.ts

```diff
@@ -91,25 +91,60 @@ describe('Arguments with @Arg', () => {
   });
 
   it('Will allow registering argument at runtime', () => {
+    @ObjectType()
+    class Foo {
+      @Field()
+      bar(baz: string, bazRequired: string): string {
+        return baz;
+      }
+    }
+
+    Arg({ type: String, isNullable: true })(Foo.prototype, 'bar', 0);
+    Arg({ type: String, isNullable: false })(Foo.prototype, 'bar', 1);
+
+    const [bazArg, bazRequiredArg] = compileObjectType(
+      Foo,
+    ).getFields().bar.args;
+
+    expect(bazArg.type).toBe(GraphQLString);
+    expect(bazRequiredArg.type).toEqual(new GraphQLNonNull(GraphQLString));
+  });
+
+  it('Respects unset defaultValue @Arg option', () => {
     @ObjectType()
     class Foo {
       @Field()
       bar(
+        @Arg({ isNullable: true })
         baz: string,
+        @Arg({ isNullable: false })
         bazRequired: string,
       ): string {
         return baz;
       }
     }
-
-    Arg({type: String, isNullable: true})(Foo.prototype, 'bar', 0);
-    Arg({type: String, isNullable: false})(Foo.prototype, 'bar', 1);
-
     const [bazArg, bazRequiredArg] = compileObjectType(
       Foo,
     ).getFields().bar.args;
-
     expect(bazArg.type).toBe(GraphQLString);
+    expect(bazArg.defaultValue).toBe(undefined);
     expect(bazRequiredArg.type).toEqual(new GraphQLNonNull(GraphQLString));
   });
+
+  it('Respects defaultValue @Arg option', () => {
+    @ObjectType()
+    class Foo {
+      @Field()
+      bar(
+        @Arg({ isNullable: true, defaultValue: 'default' })
+        baz: string,
+      ): string {
+        return baz;
+      }
+    }
+    const compiledObject = compileObjectType(Foo);
```

Comment:
Would be good to test it basing on resolving some field that is based on knowing default value.

Reply:
Yeah I couldn't think of a way to do this, maybe defining a schema in the test, compiling it and executing against it?

Reply:
Added such a condition, I hope async/await is OK, seemed the easiest way to do it.